[GTK][WPE] Crash when ForceLinear is set but linear format is unsupported

psaavedra opened the pull request:
#### cf34da0d71d1163d5f455c1d5dfe312fa0d0ea6a
<pre>
[GTK][WPE] Crash when ForceLinear is set but linear format is unsupported
<a href="https://bugs.webkit.org/show_bug.cgi?id=296034">https://bugs.webkit.org/show_bug.cgi?id=296034</a>

Reviewed by Nikolas Zimmermann.

Enforce crash when the ForceLinear flag is set but the negotiated
format does not support DRM_FORMAT_MOD_LINEAR.

This change ensures strict compliance with the user intent when
WEBKIT_USE_LINEAR_TILE_TEXTURES=1 is set and avoids silent fallback to
OpenGL or other legacy paths.

* Source/WebCore/platform/graphics/gbm/MemoryMappedGPUBuffer.cpp:
(WebCore::MemoryMappedGPUBuffer::create):
* Source/WebCore/platform/graphics/texmap/BitmapTexture.cpp:
(WebCore::BitmapTexture::updateContents):

Canonical link: <a href="https://commits.webkit.org/299153@main">https://commits.webkit.org/299153@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/181da4ad248e908f6272095314e3f1c4490893ec

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/118049 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/37727 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/28363 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/124199 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/70085 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/119927 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/38420 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/46309 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/89583 "Passed tests") | [💥 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/59185 "An unexpected error occured. Recent messages:Printed configuration") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/121001 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/30602 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/105851 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/70075 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/29670 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/23968 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/67864 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/100025 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/24146 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/127280 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/44952 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/33873 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/98255 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/45313 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/102075 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/98041 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24936 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/43441 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/21430 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/41388 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/44824 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/50498 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/44284 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/47629 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/45973 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/7b2b625853f20c9663f029245515fdcc3ec104ec. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48106

psaavedra commented:
@nikolaszimmermann thanks for the r+. However, lately I've notice those logs (LOG_ERROR)before the CRASH() only are show in webkit debug releases. Since this information is relevant to narrow down linear buffer issues when this is not supported by the platform, if you don't mind, I will replace the log message with `WTFLogAlways` so the error information will printed in the stderr output.

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/181da4ad248e908f6272095314e3f1c4490893ec. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/48106

psaavedra committed:
[GTK][WPE] Crash when ForceLinear is set but linear format is unsupported
https://bugs.webkit.org/show_bug.cgi?id=296034

Reviewed by Nikolas Zimmermann.

Enforce crash when the ForceLinear flag is set but the negotiated
format does not support DRM_FORMAT_MOD_LINEAR.

This change ensures strict compliance with the user intent when
WEBKIT_USE_LINEAR_TILE_TEXTURES=1 is set and avoids silent fallback to
OpenGL or other legacy paths.

* Source/WebCore/platform/graphics/gbm/MemoryMappedGPUBuffer.cpp:
(WebCore::MemoryMappedGPUBuffer::create):
* Source/WebCore/platform/graphics/texmap/BitmapTexture.cpp:
(WebCore::BitmapTexture::updateContents):

Canonical link: https://commits.webkit.org/299153@main
